### DISCUSSION TITLE
reroot the tree?

### DISCUSSION THREAD

**Discussion by Hongjinwu** [tskit-dev/tskit#2324] [General]
Is there a way to reroot the tree in tree sequence?

**Comment by petrelharp** [comment]
There is not. It wouldn't be too hard to build a new single-tree tree sequence that is the re-rooting of a given tree, but I'm not sure even if re-rooting of a more-than-one-tree tree sequence is well-defined? What did you have in mind?

**Comment by jeromekelleher** [comment]
Nodes in a tree also have defined times rather than branch lengths, so rerooting mainly involves making decisions about these. As @petrelharp says, if you let us know what application you have in mind we can have a think.

**Comment by Hongjinwu** [comment]
I need to manipulate a lot of trees sometime reroot is needed. because tskit is fast so I wander if that can be done based on tskit. Anyway, thanks a lot for answering.

**Comment by jeromekelleher** [comment]
Converting to a discussion, as this isn't a feature we're going to implement in tskit (most likely) but we may want to discuss the idea further.
